### PR TITLE
Restrict state.orchestrate_single to pass a pillar value if it exists

### LINKED ISSUE #61092: [BUG] salt.wait_for_event does not accept argument "pillar"
**Description**

It is not possible to use  `salt.wait_for_event` together with `salt-run state.orchestrate_single` because the latter passes a `pillar` argument to the former. The former does not accept `**kwargs` and `pillar` is not in the function signature.

**Setup**

-  VM (KVM CentOS 8 with latest Salt from repo.saltproject.io/#rhel)
  
**Steps to Reproduce the behavior**
```
[root@upstream-master ~]# salt-run state.orchestrate_single fun=salt.wait_for_event name='/salt/minion/*/start' id_list=minion1                                                                                                             
                                                                                                                                                                                                                                              
Passed invalid arguments: 'pillar' is an invalid keyword argument for 'salt.wait_for_event'                                                                                                                                                   
                                                                                                                                                                                                                                              
Usage:                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                              
    Execute a single state orchestration routine                                                                                                                                                                                              
                                                                                                                                                                                                                                              
    .. versionadded:: 2015.5.0                                                                                                                                                                                                                
                                                                                                                                                                                                                                              
    CLI Example:                                                                                                                                                                                                                              
                                                                                                                                                                                                                                              
    .. code-block:: bash                                                                                                                                                                                                                      
                                                                                                                                                                                                                                              
        salt-run state.orchestrate_single fun=salt.wheel name=key.list_all         
```


**Expected behavior**
`salt-run state.orchestrate_single fun=salt.wait_for_event name='/salt/minion/*/start' id_list=minion1` runs `salt.wait_for_event` without an error.


**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
[root@upstream-master ~]# salt --versions-report
Salt Version:
          Salt: 3003.3
 
Dependency Versions:
          cffi: 1.11.5
      cherrypy: Not Installed
      dateutil: 2.6.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: 0.35.2
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.14
      pycrypto: Not Installed
  pycryptodome: Not Installed
        pygit2: Not Installed
        Python: 3.6.8 (default, Apr 16 2020, 01:36:27)
  python-gnupg: Not Installed
        PyYAML: 3.12
         PyZMQ: 19.0.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: centos 8 Core
        locale: UTF-8
       machine: x86_64
       release: 4.18.0-193.6.3.el8_2.x86_64
        system: Linux
       version: CentOS Linux 8 Core
```
</details>

### PR DESCRIPTION
**Pull request by agraul**
### What does this PR do?

By accepting any kwargs, even if they aren't used, salt.wait_for_event can be used with state.orchestrate_single. The latter passes a "pillar" argument which raises an error without this patch.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/61092

### Previous Behavior
```
[root@upstream-master ~]# salt-run state.orchestrate_single fun=salt.wait_for_event name='/salt/minion/*/start' id_list='[minion1]'

Passed invalid arguments: 'pillar' is an invalid keyword argument for 'salt.wait_for_event'

Usage:

    Execute a single state orchestration routine   
   
    .. versionadded:: 2015.5.0   
                                                 
    CLI Example:     

    .. code-block:: bash                                                                   

        salt-run state.orchestrate_single fun=salt.wheel name=key.list_all  
```
### New Behavior
```
[root@upstream-master ~]# salt-run state.orchestrate_single fun=salt.wait_for_event name='/salt/minion/*/start' id_list='[minion1]' 
# waits
```

### Merge requirements satisfied?

- [x] Docs: Just a bug fix, does not need docs changes IMO.
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated: Not yet, old test is not moved to pytest yet. I want to wait for feedback on the fix before migrating the test.

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/states/saltmod.py [hunk 1]
```diff
@@ -608,7 +608,7 @@ def function(
     return func_ret
 
 
-def wait_for_event(name, id_list, event_id="id", timeout=300, node="master"):
+def wait_for_event(name, id_list, event_id="id", timeout=300, node="master", **kwargs):
```
[the remark this answers is not in the document]
Comment (agraul) [reply salt/states/saltmod.py]: I fixed it in `orchestrate_single`, at least for this use case. When `pillar=None`, it does not make sense to pass it to `state.single`, which only looks for it with `pillar_override = kwargs.get("pillar")`. In English, passing `pillar=None` and not passing `pillar` at all does the same thing.

Now I wonder what the best type of test for this is. Is a unit test for `runners.state.orchestrate_single` enough? I'm thinking of a call to it without a pillar and then check that it's not in the `kwargs` received by a mocked `state.single`. What do you think?

**Comment by agraul** [comment]
I've added a pair of unit tests. Please let me know if this is not the correct approach to test the change.

@s0undt3ch could you review again please?

**Review by meaksh — DISMISSED** [review]
This looks good to me :+1: Thanks

**Review by garethgreenaway — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]

File: tests/pytests/unit/runners/test_state.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/python3
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/runners/test_state.py:2]:
```suggestion

```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/runners/test_state.py (hunk 2 above)
Comment (agraul) [reply tests/pytests/unit/runners/test_state.py:2]: Good catch, I should turn off the python file template that inserts the shebang

**Comment by Ch3LL** [comment]
looks like there is just the pre-commit test failure and then we should be able to get this one in

**Comment by agraul** [comment]
Both failing tests time out, how do we deal with that?

**Comment by Ch3LL** [comment]
I went ahead and restarted the tests

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**